src/kayenta/edit/filterTemplatesValidation.ts

```diff
@@ -0,0 +1,94 @@
+import { get, isEmpty } from 'lodash';
+
+import { ICanaryMetricConfig } from '../domain';
+import { IEditingTemplateState } from '../reducers/editingTemplate';
+
+export interface ICanaryFilterTemplateValidationMessage {
+  message: string;
+}
+
+export interface ICanaryFilterTemplateValidationMessages {
+  errors: { [key: string]: ICanaryFilterTemplateValidationMessage };
```

Comment:
You could strongly type the keys here (if there are only going to be two, maybe `errors: { [key: 'templateName' | 'templateValue']: ICanaryFilterTemplateValidationMessage}`?)